db/collection.go

```diff
@@ -170,6 +172,37 @@ func (db *db) createCollection(
 	return db.getCollectionByID(ctx, txn, desc.ID)
 }
 
+// validateCollectionDefinitionPolicyDesc validates that the policy definition is valid, beyond syntax.
+//
+// Ensures that the information within the policy definition makes sense,
+// this function might also make relevant remote calls, like call the acp module.
+func (db *db) validateCollectionDefinitionPolicyDesc(
+	ctx context.Context,
+	policyDesc immutable.Option[client.PolicyDescription],
```

Comment:
suggestion: I don't think `Option` is needed, as it should be the callers responsibility to determine if there exists a policy, and to apply validation. This function is only responsible for the *actual* validation of said policy.

Reply:
I am ok either way. Leaving as is unless stronger pref or others also want this.

Reply:
Im OK to leave it as is. This is different from the `permission` util stuff that I *do* think should be changed to remove the `Option` since this is a method on the `db` and is related to the `CollectionDescription` stuff which does use the `Option` so theres no conceptual gap to cross.